code/modules/mob/living/carbon/human/human_attackhand.dm

```diff
@@ -11,6 +11,16 @@
 		return
 
 	..()
+	if ((H.invisibility == INVISIBILITY_LEVEL_TWO) && back && (istype(back, /obj/item/weapon/rig)))

```

Comment:
If this is supposed to affect the H, and not the src, then you'd need to make the checks be `H.back` and `istype(H.back, /obj/item/weapon/rig)`, otherwise it'll check the source.